tez-common/src/main/java/org/apache/tez/common/web/ProfileServlet.java

```diff
@@ -0,0 +1,326 @@
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class ProfileServlet extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+    private static final Logger LOG = LoggerFactory.getLogger(ProfileServlet.class);
+    private static final String ACCESS_CONTROL_ALLOW_METHODS = "Access-Control-Allow-Methods";
+    private static final String ALLOWED_METHODS = "GET";
+    private static final String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+    private static final String CONTENT_TYPE_TEXT = "text/plain; charset=utf-8";
+    private static final String ASYNC_PROFILER_HOME_ENV = "ASYNC_PROFILER_HOME";
+    private static final String ASYNC_PROFILER_HOME_SYSTEM_PROPERTY = "async.profiler.home";
+    private static final String PROFILER_SCRIPT = "/profiler.sh";
+    private static final int DEFAULT_DURATION_SECONDS = 10;
+    private static final AtomicInteger ID_GEN = new AtomicInteger(0);
+    public static final String OUTPUT_DIR = System.getProperty("java.io.tmpdir") + "/prof-output";
+
+    enum Event {
+        CPU("cpu"),
+        ALLOC("alloc"),
+        LOCK("lock"),
+        PAGE_FAULTS("page-faults"),
+        CONTEXT_SWITCHES("context-switches"),
+        CYCLES("cycles"),
+        INSTRUCTIONS("instructions"),
+        CACHE_REFERENCES("cache-references"),
+        CACHE_MISSES("cache-misses"),
+        BRANCHES("branches"),
+        BRANCH_MISSES("branch-misses"),
+        BUS_CYCLES("bus-cycles"),
+        L1_DCACHE_LOAD_MISSES("L1-dcache-load-misses"),
+        LLC_LOAD_MISSES("LLC-load-misses"),
+        DTLB_LOAD_MISSES("dTLB-load-misses"),
+        MEM_BREAKPOINT("mem:breakpoint"),
+        TRACE_TRACEPOINT("trace:tracepoint"),
+        ;
+
+        private final String internalName;
+
+        Event(final String internalName) {
+            this.internalName = internalName;
+        }
+
+        public String getInternalName() {
+            return internalName;
+        }
+
+        public static Event fromInternalName(final String name) {
+            for (Event event : values()) {
+                if (event.getInternalName().equalsIgnoreCase(name)) {
+                    return event;
+                }
+            }
+            return null;
+        }
+    }
+
+    enum Output {
+        SUMMARY,
+        TRACES,
+        FLAT,
+        COLLAPSED,
+        SVG,
+        TREE,
+        JFR
+    }
+
+    private final Lock profilerLock = new ReentrantLock();
+    private Integer pid;
+    private String asyncProfilerHome;
+    private transient Process process;
+
+    public ProfileServlet() {
+        this.asyncProfilerHome = getAsyncProfilerHome();
+        this.pid = getPid();
+        LOG.info("Servlet process PID: {} asyncProfilerHome: {}", pid, asyncProfilerHome);
+    }
+    private Integer getPid(){
```

Comment:
please refactor this to a public static utility method, e.g. to TezUtilsInternal